metrics.py

```diff
@@ -427,17 +439,25 @@ def calculate_sample_statistics(self, networks_sample: np.ndarray) -> np.ndarray
         if self.requires_graph:
             networks_as_graphs = [connectivity_matrix_to_G(W, self.is_directed) for W in networks_sample]
 
+        networks_as_sparse_tensor = np_tensor_to_sparse_tensor(networks_sample)
+
         feature_idx = 0
         for metric in self.metrics:
             n_features_from_metric = metric.get_effective_feature_count(self.n_nodes)
 
             if metric.requires_graph:
                 networks = networks_as_graphs
             else:
-                networks = networks_sample
-
-            features_of_net_samples[feature_idx:feature_idx + n_features_from_metric] = metric.calculate_for_sample(
-                networks)
+                # networks = networks_sample
+                networks = networks_as_sparse_tensor
+                
+            features = metric.calculate_for_sample(networks)
+            if isinstance(features, torch.Tensor):
+                if features.is_sparse:
+                    features = features.to_dense()
+                features = features.numpy()
+
+            features_of_net_samples[feature_idx:feature_idx + n_features_from_metric] = features
             feature_idx += n_features_from_metric
 
         return features_of_net_samples
```

Comment:
Do we use again the sample itself during optimization after we calculate the features? Maybe we can get rid of the sparse sample here as well